source/gems/dry-validation/comparison-with-activemodel.html.md

```diff
@@ -0,0 +1,589 @@
+```
+
+####Associations
+
+If you want to be sure that an association is absent, you'll need create a custom predicate to test whether the associated object itself is absent.
+
+> TODO: give an example of what such a predicate could looklike
+
+**Booleans**
+
+To validate the absence of a boolean field (e.g. not true or false) you should use:
+
+```ruby
+required(:attr).filled(excluded_from?: [true, false])
+```
+
+**Booleans**
+
+To validate the absence of a boolean field (e.g. not true or false) you should use:
+
+`required(:attr).filled(:none?)`
+
+### 2.11 uniqueness
+
+Rails' `uniqueness` validation is fundamentally different from the other validations because it requires a query against a database. (Accordingly, the uniqueness validation is contained within the `activerecord` gem, while other validations are part of `activemodel`.) TODO does this mean that dry-v can't handle this case? expand
+
+### 2.12 validates_with
+
+TODO expand
+
+### 2.13 validates_each
+
+TODO expand
+
+### 3. Common Validation Options
+
+These are the common options allowed by ActiveModel validations, and their equivalents in dry-validation
+
+**3.1 `:allow_nil`**
+
+Simply use `maybe` instead of `required` when defining your rules.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, length: { minimum: int, allow_nil: true }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(int?, min_size?: int)
+```
+
+**3.2  `:allow_blank`**
+
+In dry-validation you will need to use a block when defining your rule instead of `filled`, and include the `.empty?` predicate into your rule.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, length: { minimum: int, allow_blank: true }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr) { empty? | int? & min_size?: int )
+```
+
+
+**3.3 `:message`**
+
+Custom messages are implemented through a separate YAMl file. See [Error Messages](http://dry-rb.org/gems/dry-validation/error-messages/) for full instructions.
+
+**3.4 `:on`**
+
+In dry-validation, validations are defined in schemas. You can create separate schemas for various states (e.g UserCreateSchema, UserUpdateSchema) and then choose the correct schema to run in the relevant action.
+
+You can keep your schema code nice and DRY by [reusing schemas](gems/dry-validation/reusing-schemas/).
+
+### 4. Conditional Validation
+In ActiveModel you can use `:if` or `:unless` to only perform a validation based on the result of a proc or method.
+
+A simple schema can look like this:
+
+```ruby
+validates :card_number, presence: true, if: :paid_with_card?
+
+def paid_with_card?
+  payment_type == "card"
+end
+```
+
+To achieve this in dry-validation you can use [high-level rules](/gems/dry-validation/high-level-rules/).
+
+.1. Declare a rule for each of the attributes you need to reference:
+
+```ruby
+required(:payment_type).filled(included_in?: ["card", "cash", "cheque"])
+required(:card_number).maybe
```

Comment:
Pretty sure that we have to do something with maybe even if it is just `required(:card_number).maybe(:str?)`